app/Http/Controllers/TrainingController.php

```diff
@@ -230,7 +231,8 @@ public function store(Request $request)
             'notes' => isset($data['comment']) ? 'Comment from application: '.$data['comment'] : '',
             'motivation' => isset($data['motivation']) ? $data['motivation'] : '',
             'experience' => isset($data['experience']) ? $data['experience'] : null,
-            'english_only_training' => key_exists("englishOnly", $data) ? true : false
+            'english_only_training' => key_exists("englishOnly", $data) ? true : false,
+            'type' => $data['type']
```

Comment:
This breaks the creation of applications done by students (the form, not the manual creation you made)
```
[2021-05-03 15:47:16] local.ERROR: Undefined index: type {"userId":10000010,"exception":"[object] (ErrorException(code: 0): Undefined index: type at D:\\GitHub\\controlcenter\\app\\Http\\Controllers\\TrainingController.php:235)
[stacktrace]
#0 D:\\GitHub\\controlcenter\\app\\Http\\Controllers\\TrainingController.php(235): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(8, 'Undefined index...', 'D:\\\\GitHub\\\\contr...', 235, Array)
#1 D:\\GitHub\\controlcenter\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Controller.php(54): App\\Http\\Controllers\\TrainingController->store(Object(Illuminate\\Http\\Request))
#2 D:\\GitHub\\controlcenter\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction('store', Array)
#3 D:\\GitHub\\controlcenter\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Route.php(254): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(App\\Http\\Controllers\\TrainingController), 'store')
#4 D:\\GitHub\\controlcenter\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Route.php(197): Illuminate\\Routing\\Route->runController()
#5 D:\\GitHub\\controlcenter\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(695): Illuminate\\Routing\\Route->run()```

Reply:
solved in commit d47dc26be51713859ddc3c019152eeb8d55d86c4

Reply:
That won't solve it. `type` is an integer, `isset()` returns a boolean. It should be something like `isset($data['type']) ? $data['type'] : 1` since type 1 is a standard trianing.